Zend/zend_compile.c

```diff
@@ -848,6 +848,9 @@ uint32_t zend_add_member_modifier(uint32_t flags, uint32_t new_flag) /* {{{ */
 			"Cannot use the final modifier on an abstract class member", 0);
 		return 0;
 	}
+	if ((new_flags & ZEND_ACC_PRIVATE) && (new_flags & ZEND_ACC_FINAL)) {
+		zend_error(E_COMPILE_WARNING, "Private methods are inherently final as they cannot be overridden by other classes");
```

Comment:
I don't like the wording here. It's more a matter of "cannot be final" than "is always implicitly final". Querying a private method in reflection does not report it as final.

Reply:
I've updated the wording of the warning.